Watch Extension/State Management/FeedingReducer.swift

```diff
@@ -1,16 +1,64 @@
 import Common
 import Swift
+import WatchConnectivity
+
+typealias FeedingState = (
+    feedings: [Feeding],
+    showCommunicationFyiDialog: Bool,
+    showSavedFyiDialog: Bool,
+    isLoading: Bool
+)
 
 enum FeedingAction {
     case update(feedings: [Feeding])
+    case communicate(type: FeedingType, side: FeedingSide, action: Common.FeedingAction)
+    case communicationFailed
+    case showFeedingStopped
+    case loadingFinished
 }
 
-func feedingReducer(value: inout [Feeding], action: FeedingAction) -> [Effect<FeedingAction>] {
+func feedingReducer(feedingState: inout FeedingState, action: FeedingAction) -> [Effect<FeedingAction>] {
     switch action {
     case let .update(feedings):
         // TODO: show all after formatting the list view UI better
         // This will also reqire better handleing pause/resume/stop reducer actions
-        value = feedings.filter { !$0.isFinished }
+        feedingState.feedings = feedings.filter { !$0.isFinished }
+        return []
+    case let .communicate(type, side, action):
+        let info = WatchFeedingCommunication(type: type, side: side, action: action)
+        let jsonEncoder = JSONEncoder()
+        guard let d = try? jsonEncoder.encode(info), WCSession.default.isReachable else {
+            feedingState.showCommunicationFyiDialog = true
+            return []
+        }
+        feedingState.isLoading = true
+        return [
+            { callback in
+                WCSession.default.sendMessageData(
+                    d,
+                    replyHandler: { _ in
+                        defer { callback(.loadingFinished) }
+                        switch action {
+                        case .start: break
+                        case .stop: callback(.showFeedingStopped)
+                        case .pause: break
+                        case .resume: break
+                        }
+                },
+                    errorHandler: { error in
+                        print("Error sending the message: \(error.localizedDescription)")
+                        callback(.communicationFailed)
+                })
+            }
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/jasonzurita/BabyPatterns/blob/d0969e50f3967ea66270aaf4bf0f82ba44576d07/Watch Extension/State Management/FeedingReducer.swift#L52"></span>Multi-line collection literals should have trailing commas.<br />
<code>trailing_comma</code> <code>FeedingReducer.swift:52</code></td>
    </tr>
  </tbody>
</table>